src/components/useBoard.js

```diff
@@ -0,0 +1,85 @@
+import { useState, useEffect } from 'react';
+import { getRandomNumber, loadCell, switchCell } from '../utils';
+
+const useBoard = wordLengths => {
+  // create an empty board
+  const [squares, setSquares] = useState([]);
+  const [wordNo, setWordNo] = useState(1);
+
+  useEffect(() => {
+    let workSquares = JSON.parse(JSON.stringify(squares));
+    let posY = 1;
+    let workWordNo = wordNo;
+    let posX = 1;
+    let alignment = 'row';
+    let randomNumber;
+    // outer loop runs through selected word lengths
+    for (let i = 0; i < wordLengths.length; i++) {
+      // check to see if you have made the largest board possible yet
+      if (alignment === 'row' && posX + wordLengths[i] > 9) {
+        break;
+      } else if (alignment === 'column' && posY + wordLengths[i] > 9) {
+        break;
+      }
+      // inner loop processes each letter in a word
+      randomNumber = getRandomNumber(1, 2);
+      let doubleWord = false;
+      for (let x = 1; x < wordLengths[i]; x++) {
+        // check if letter is used in 2 words
+        if ((x === 1 && i !== 0) || (x === 1 && posX !== 1)) {
```

Comment:
here's another one of those if/else that can just be turned into a single line.
`doubleWord = (x === 1 && i !== 0) || (x === 1 && posX !== 1)`

Reply:
Done